### PR TITLE
test and build wheels for Py3.{7,8,9,10}

### PR DESCRIPTION
**Pull request by mpenkov**

### CODE REVIEW

**Comment by mpenkov** [comment]
Oof, finally got everything to build for Python 3.10.

There are some catches:

- Had to jump through hoops to upgrade pip for Py3.10, so there are some hacky scripts
- Gave up on running unit tests using the wheel, because some blinking tests can fail the build. I'm using a much simpler test now: install the wheel, import gensim, print its version.

Please let me know what you think.

**Comment by piskvorky** [comment]
The simpler the better :)

**Review by piskvorky — APPROVED** [review]

**Comment by mpenkov** [comment]
OK, I re-enabled the tests and increased the tolerance for the test that was failing. It works, so I'm merging this now.

**Comment by gojomo** [comment]
Sorry, I thought my 2 code review comments had been posted days ago – they were visible to me, even outside the 'changed files' page – but the `pending` notation alongside them, that I thought meant *you* hadn't yet dismissed, actually meant I hadn't yet submitted the batch for wider visibility. Ugh. I should probably stick to "submit as individual comment".

File: .github/workflows/build-wheels.yml [hunk 1]
```diff
@@ -107,14 +102,19 @@ jobs:
...
+
     env:
       PKG_NAME: gensim
       REPO_DIR: gensim
       BUILD_COMMIT: HEAD
       PLAT: x86_64
       UNICODE_WIDTH: 32
       MB_PYTHON_VERSION: ${{ matrix.python-version }} # MB_PYTHON_VERSION is needed by Multibuild
-      TEST_DEPENDS: Morfessor==2.0.2a4 python-levenshtein==0.12.0 visdom==0.1.8.9 pytest pytest-cov mock cython nmslib pyemd testfixtures scikit-learn pyemd
+      TEST_DEPENDS: pytest mock testfixtures
```
[the remark this answers is not in the document]
Comment (mpenkov) [reply .github/workflows/build-wheels.yml:117]: It's possible, but this is for the purpose of testing the wheel, so running tests that require those dependencies isn't strictly necessary.

**Pull request merged**